File: internal/domain/sync.go
package domain

import (
	"encoding/json"
	"time"

	"github.com/google/uuid"
)

type SyncRecord struct {
	LegacyAttachment  LegacyAttachment
	OwnerClassLibrary string
	OwnerClassName    string
	MinioKey          string
	SHA256Hash        string
}

type SyncCheckpoint struct {
	TableName        string
	LastDCCheck      []byte
	LastSyncAt       time.Time
	RecordsProcessed int64
	Status           string
}

type SyncLogEntry struct {
	ID            uuid.UUID
	DocumentID    *uuid.UUID
	LegacyOwnerID string
	LegacyFileID  string
	Action        string // create, update, delete, error
	Status        string // success, failed, retrying, dlq
	ErrorMessage  string
	DurationMs    int
	SyncedAt      time.Time
}

type DLQEntry struct {
	ID            uuid.UUID
	LegacyOwnerID string
	LegacyFileID  string
	TableName     string
	RetryCount    int
	MaxRetries    int
	NextRetryAt   *time.Time
	LastError     string
	PayloadJSON   json.RawMessage
	Status        string // pending, retrying, exhausted, resolved
	CreatedAt     time.Time
	UpdatedAt     time.Time
}

File: web/src/types/sync.ts
export interface SyncStatus {
  is_running: boolean;
  last_sync_at: string | null;
  next_sync_at: string | null;
  documents_synced: number;
  documents_failed: number;
  dlq_count: number;
}

export interface DLQEntry {
  id: number;
  document_id: number;
  error_message: string;
  retry_count: number;
  max_retries: number;
  created_at: string;
  updated_at: string;
  document_title?: string;
}

export interface SyncLogEntry {
  id: number;
  sync_type: string;
  status: string;
  started_at: string;
  completed_at: string | null;
  documents_processed: number;
  documents_failed: number;
  error_message: string | null;
}
